File: generic/client/gwt/war/spike/configuration.js
CHASSIS = new Object();

// configure service URLs

CHASSIS.userDetailsServiceEndpointURL = "/chassis-generic-service-user/gwtrpc/userdetails";

CHASSIS.studyFeedURL = "/chassis-generic-service-exist/atom/edit/studies";
CHASSIS.studyQueryServiceURL = "/chassis-generic-service-exist/query/studies.xql";

CHASSIS.studyQuestionnaireURL = "/chassis-generic-client-gwt/questionnaire/study-questionnaire.xml";

CHASSIS.submissionFeedURL = "/chassis-generic-service-exist/atom/edit/submissions";
CHASSIS.submissionQueryServiceURL = "/chassis-generic-service-exist/query/submissions.xql";

CHASSIS.dataFileFeedURL = "/chassis-generic-service-exist/atom/edit/datafiles";
CHASSIS.dataFileQueryServiceURL = "/chassis-generic-service-exist/query/datafiles.xql";

// configure role name prefix
CHASSIS.userChassisRolesPrefix = "ROLE_CHASSIS_";

// configure permission suffix to chassis role mappings
CHASSIS.chassisRoleCoordinator = {"permissionSuffix":"COORDINATOR",
								  "label":{
											"en":"Coordinator"
										  }
								 };
CHASSIS.chassisRoleCurator = {"permissionSuffix":"CURATOR",
							  "label":{
										"en":"Curator"
									  }
							 };
CHASSIS.chassisRoleGatekeeper = {"permissionSuffix":"GATEKEEPER",
								  "label":{
											"en":"Gate Keeper"
										  }
								 };
CHASSIS.chassisRoleSubmitter = {"permissionSuffix":"SUBMITTER",
								  "label":{
											"en":"Submitter"
										  }
								 };
CHASSIS.chassisRoleUser = {"permissionSuffix":"USER",
							  "label":{
										"en":"User"
									  }
							 };

// configure modules
CHASSIS.modules = [
                   {
                	   "id":"clinical",
                	   "label":{
                	   		"en":"Clinical"
                   		}
                   },
                   {
                	   "id":"invitro",
                	   "label":{
                	   		"en":"In Vitro"
                   		}
                   },
                   {
                	   "id":"molecular",
                	   "label":{
                	   		"en":"Molecular"
                   		}
                   },
                   {
                	   "id":"pharmacology",
                	   "label":{
                	   		"en":"Pharmacology"
                   		}
                   }
];

File: generic/client/gwt/src/main/java/org/cggh/chassis/generic/client/gwt/configuration/client/Configuration.java
/**
 * 
 */
package org.cggh.chassis.generic.client.gwt.configuration.client;


import com.google.gwt.core.client.JsArray;

/**
 * @author aliman
 *
 */
public class Configuration {

	public static native String getUserDetailsServiceEndpointURL() /*-{
		return $wnd.CHASSIS.userDetailsServiceEndpointURL;
	}-*/;
	
	public static native String getUserChassisRolesPrefix() /*-{
	return $wnd.CHASSIS.userChassisRolesPrefix;
	}-*/;
	
	public static native String getStudyFeedURL() /*-{
		return $wnd.CHASSIS.studyFeedURL;
	}-*/;

	public static native String getSubmissionFeedURL() /*-{
		return $wnd.CHASSIS.submissionFeedURL;
	}-*/;

	/**
	 * @return
	 */
	public static native String getSubmissionQueryServiceURL()  /*-{
		return $wnd.CHASSIS.submissionQueryServiceURL;
	}-*/;

	public static native JsArray<Module> getModules() /*-{
		return $wnd.CHASSIS.modules;
	}-*/;

	/**
	 * @return
	 */
	public static native String getStudyQueryServiceURL() /*-{
		return $wnd.CHASSIS.studyQueryServiceURL;
	}-*/;
	
}
